### PR TITLE
Calendar translate

### PR DESCRIPTION
**Pull request by kaennchenstruggle**
Hello and thank you for wanting to contribute to the MagicMirror² project

**Please make sure that you have followed these 4 rules before submitting your Pull Request:**

> 1. Base your pull requests against the `develop` branch.

DONE ;D

> 2. Include these infos in the description:
> - Does the pull request solve a **related** issue?

NO
> - What does the pull request accomplish? Use a list if needed.

For calendar entries containing a year (e.g. DOB) in the title, the age can be calculated.

Example before:
grafik
after:
grafik

Achieved by adding a new keyword `transform` to customEvents
```
customEvents: [
                        {keyword: 'Geburtstag', symbol: 'birthday-cake', color: 'Gold', transform: { search: '^([^\']*) \'(\\d{4})$' , replace: '$1 ($2.)', yearmatchgroup: 2}},
                        {keyword: 'in Hamburg', transform: { search: ' in Hamburg$' , replace: ''}} 
		],
```

and therewith obsoleting `titleReplace`; a backward compatibility part is already included.

If `yearmatchgroup` is unset, behaviour is as in previous code (some additions to which RegExes are accepted, though)

If `yearmatchgroup` is set, it is considered the RegEx match group id, which will be used for calculating the age.

> - If it includes major visual changes please add screenshots.

NO

> 3. Please run `npm run lint:prettier` before submitting so that style issues are fixed.

DONE

> 4. Don't forget to add an entry about your changes to  the CHANGELOG.md file.

DONE

> Thanks again and have a nice day!

You too and if any questions, feel free to let me know.

### CODE REVIEW

File: modules/default/calendar/calendar.js
Comment (rejas) [inline modules/default/calendar/calendar.js:337]: Please use let/const instead of var

File: CHANGELOG.md
Comment (kaennchenstruggle) [inline CHANGELOG.md:20]: as titleReplace is not completely removed (backward compatibility is included in PR) but users should be encouraged to use the new style, I'm not sure if this is the right place. Perhaps you find a better one

Comment (rejas) [reply CHANGELOG.md:20]: this would probably best be documented in the [doc-repo](https://github.com/MichMich/MagicMirror-Documentation) after this PR got merged

**Comment by kaennchenstruggle** [comment]
@rejas sorry to ask but I'm absolutely new to PR's... is there anything pending form my side (check a box, resolve an issue, etc...)?

**Comment by rejas** [comment]
> @rejas sorry to ask but I'm absolutely new to PR's... is there anything pending form my side (check a box, resolve an issue, etc...)?

Writing a test for your feature would be nice but its not a must. Adding a PR for the documentation repo would be needed once this is merged.

**Comment by rejas** [comment]
Could you also check this:

"Check failure on line 119 in modules/default/calendar/calendarutils.js
GitHub
Actions
/ test (20.x)

Unexpected string concatenation"

**Comment by rejas** [comment]
Seems that your changes make the
  ● Calendar utils tests › titleTransform and shorten combined › should replace the birthday and wrap nicely
  fail, could you check it out (ignore the other failing tests for now)

**Comment by kaennchenstruggle** [comment]
> Seems that your changes make the ` ● Calendar utils tests › titleTransform and shorten combined › should replace the birthday and wrap nicely` fail, could you check it out (ignore the other failing tests for now)

Adjusted it, I hope it works out as I can't seem to get jest working on my system

**Comment by rejas** [comment]
> Adjusted it, I hope it works out as I can't seem to get jest working on my system

I did fix the linting issues but the tests still fail. How doesnt jest work on your system?

**Comment by kaennchenstruggle** [comment]
> > Adjusted it, I hope it works out as I can't seem to get jest working on my system
> 
> I did fix the linting issues but the tests still fail. How doesnt jest work on your system?

Thanks for fixing, no clue about jest. I think I screwed up apt for whatever reason.

**Comment by rejas** [comment]
> Thanks for fixing, no clue about jest. I think I screwed up apt for whatever reason.

Can you psot the errors you get? normally the testing framework gets inspalled via npm, not apt.

your PR looks fine now, will merge it once I figure out the other failing test (unrelated to your changes)

**Review by rejas — APPROVED** [review]

**Comment by kaennchenstruggle** [comment]
> > Thanks for fixing, no clue about jest. I think I screwed up apt for whatever reason.
> 
> Can you psot the errors you get? normally the testing framework gets inspalled via npm, not apt.
> 
> your PR looks fine now, will merge it once I figure out the other failing test (unrelated to your changes)

Thank you so much for your help on my first PR.

And thanks for pointing out the npm part, at least jest runs now. Still throwing some errors:

```
 FAIL  tests/e2e/modules/calendar_spec.js
  â— Test suite failed to run

    SyntaxError: Bad control character in string literal in JSON at position 184320
        at JSON.parse (<anonymous>)

    > 1 | const jsdom = require("jsdom");
        |                               ^
```

**Comment by kaennchenstruggle** [comment]
And also updated doc-repo (develop) with PR 207 https://github.com/MichMich/MagicMirror-Documentation/pull/207 hope this is okay.

**Pull request merged**
